dora/src/bytecode/generator.rs

```diff
@@ -932,6 +934,11 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
                 BinOp::Cmp(CmpOp::Ge) => self.gen.emit_test_ge_char(dest, lhs_reg, rhs_reg),
                 _ => unreachable!(),
             },
+            Intrinsic::EnumEq | Intrinsic::EnumNe => match op {
```

Comment:
Is there a reason that there is an explicit `Intrinsic::EnumNe`? This seems to be redundant, as we also have the operand.

Reply:
I think you need to respect the intrinsic here, not the operand.

Reply:
Changed.